app/src/main/java/ru/practicum/android/diploma/ui/industry/IndustryFragment.kt

```diff
@@ -4,25 +4,154 @@ import android.os.Bundle
+    }
+
+    private fun showLoading() {
+        binding.progressBar.isVisible = true
+        binding.centralImageHolder.isVisible = false
+        binding.recyclerView.isVisible = false
+        binding.stateTextView.isVisible = false
+        binding.selectButton.isVisible = false
+    }
+
+    private fun showEmpty(message: String) {
+        showImageAndTextState()
+        binding.centralImageHolder.setImageResource(R.drawable.empty_list_image)
+        binding.stateTextView.text = message
+    }
+
+    private fun showError(errorMessage: String) {
+        showImageAndTextState()
+        binding.centralImageHolder.setImageResource(R.drawable.error_image)
+        binding.stateTextView.text = errorMessage
+    }
+
+    private fun showImageAndTextState() {
+        binding.progressBar.isVisible = false
+        binding.centralImageHolder.isVisible = true
+        binding.recyclerView.isVisible = false
+        binding.stateTextView.isVisible = true
+        binding.selectButton.isVisible = false
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentIndustryBinding.inflate(
+            inflater,
+            container,
+            false
+        )
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         binding.topAppBar.setNavigationOnClickListener {
             findNavController().popBackStack()
         }
+
+        industryAdapter = IndustryAdapter {
+            binding.selectButton.isVisible = true
+        }
+
+        binding.recyclerView.layoutManager = LinearLayoutManager(requireContext())
+        binding.recyclerView.adapter = industryAdapter
+        binding.recyclerView.animation = null
+
+        binding.inputEditText.addTextChangedListener(onTextChanged = {
+                charSequence: CharSequence?,
+                i: Int,
+                i1: Int,
+                i2: Int ->
+            if (charSequence?.isNotEmpty() == true) {
+                binding.searchFrame.setEndIconDrawable(R.drawable.close_icon)
+                industryViewModel.searchDebounce(changedText = charSequence.toString())
+                binding.searchFrame.setEndIconOnClickListener {
+                    binding.inputEditText.setText(getString(R.string.empty_string))
+                    industryViewModel.loadIndustries()
+                }
+            } else {
+                binding.searchFrame.setEndIconDrawable(R.drawable.search_icon)
+                binding.searchFrame.clearOnEndIconChangedListeners()
+            }
+        })
+        binding.inputEditText.setOnEditorActionListener { textView, actionId, keyEvent ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                industryViewModel.searchDebounce(changedText = textView.text.toString())
+                true
+            }
+            false
+        }
```

Comment:
Что-то как-то сложно, почему не сделать как при поиске вакансий?
```suggestion
            override fun afterTextChanged(s: Editable?) {
                if (s?.isNotEmpty() == true) {
                    binding.searchFrame.setEndIconDrawable(R.drawable.close_icon)
                    viewModel.searchDebounce(changedText = s.toString())
                } else {
                    binding.searchFrame.setEndIconDrawable(R.drawable.search_icon)
                }
            }
```